MatchServer/MMatchRuleGunGame.cpp

```diff
@@ -0,0 +1,134 @@
+#include "stdafx.h"
+#include "MMatchRuleGunGame.h"
+#include "MBlobArray.h"
+
+MMatchRuleGunGame::MMatchRuleGunGame(MMatchStage* pStage) : MMatchRule(pStage)
+{
+}
+
+MMatchRuleGunGame::~MMatchRuleGunGame()
+{
+}
+
+void MMatchRuleGunGame::OnBegin()
+{
+	if (m_pGGSet.size() > 0) {
+		for (auto itor = m_pGGSet.begin(); itor != m_pGGSet.end(); ++itor)
+		{
+			delete *itor;
+		}
+		m_pGGSet.clear();
+	}
+
+	int randSet = rand() % MGetGunGame()->GetSetCount();
```

Comment:
Yeah, ill change some of the variables around. I get what you're saying now, il change it to vector<vector>. I was wondering, do you have a skype just for this project? Would love to talk with you about your ideas surrounding this project.